ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -305,8 +311,10 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		}
 	}
 
+	deployStatus := rayJobInstance.Status.JobDeploymentStatus
+	isDeployStatusRunningOrFailedToGetStatus := deployStatus == rayv1.JobDeploymentStatusRunning || deployStatus == rayv1.JobDeploymentStatusFailedToGetJobStatus
```

Comment:
Why do we need `deployStatus == rayv1.JobDeploymentStatusFailedToGetJobStatus` if we have already updated the `JobDeploymentStatus` above?

Reply:
I'm just being defensive here, I don't have a concrete situation in mind. I agree it shouldn't happen anymore (since in the case of `FailedToGetJobStatus`, we requeue before reaching this line).

Reply:
We should remove it if it is not necessary.

Reply:
Removed 9ed53b0